packages/abi-coder/src/encoding/strategies/getCoderV1.ts

```diff
@@ -122,21 +123,26 @@ export const getCoder: GetCoderFn = (
     return new VecCoder(itemCoder as Coder);
   }
 
-  const structMatch = structRegEx.exec(resolvedAbiType.type)?.groups;
+  const structMatch = structRegEx.test(resolvedAbiType.type);
   if (structMatch) {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const [name] = resolvedAbiType.type.match(lastNameRegExMatch)!;
     const coders = getCoders(components, { getCoder });
-    return new StructCoder(structMatch.name, coders);
+    return new StructCoder(name, coders);
   }
 
-  const enumMatch = enumRegEx.exec(resolvedAbiType.type)?.groups;
+  const enumMatch = enumRegEx.test(resolvedAbiType.type);
   if (enumMatch) {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const [name] = resolvedAbiType.type.match(lastNameRegExMatch)!;
+
     const coders = getCoders(components, { getCoder });
 
     const isOptionEnum = resolvedAbiType.type === OPTION_CODER_TYPE;
```

Comment:
```suggestion
    const isOptionEnum = resolvedAbiType.type.test(optionCoderRegex);
```

Do we want to be testing against the regex here?

Reply:
No, at this point we are only interested in the standard `Option`.